src/cc_catalog_airflow/dags/provider_api_scripts/stocksnap.py

```diff
@@ -0,0 +1,204 @@
+        endpoint,
+        retries,
+        query_params,
+        headers=headers
+    )
+    if response_json is None:
+        return None
+    else:
+        data = response_json.get("results")
+        return data
+
+
+def _process_item_batch(items_batch):
+    for item in items_batch:
+        item_meta_data = _extract_item_data(item)
+        if item_meta_data is None:
+            continue
+        image_store.add_item(**item_meta_data)
+    return image_store.total_items
+
+
+def _extract_item_data(media_data):
+    """
+    Extract data for individual image
+    """
+    try:
+        foreign_id = media_data["img_id"]
+        foreign_landing_url, page = _get_foreign_landing_page(foreign_id)
+    except (TypeError, KeyError, AttributeError):
+        logger.info("Foreign landing url not resolved.")
+        logger.info(f"{json.dumps(media_data, indent=2)}")
+        return None
+    image_url, width, height = _get_image_info(media_data)
+    if image_url is None:
+        logger.info("Found no image url.")
+        logger.info(f"{json.dumps(media_data, indent=2)}")
+        return None
+    item_license = _get_license()
+    title = _get_title(page)
+    creator, creator_url = _get_creator_data(page)
+    thumbnail = image_url
+    metadata = _get_metadata(media_data)
+    tags = _get_tags(media_data)
+
+    return {
+        'title': title,
+        'creator': creator,
+        'creator_url': creator_url,
+        'foreign_identifier': foreign_id,
+        'foreign_landing_url': foreign_landing_url,
+        'image_url': image_url,
+        'height': height,
+        'width': width,
+        'thumbnail_url': thumbnail,
+        'license_info': item_license,
+        'meta_data': metadata,
+        'raw_tags': tags
+    }
+
+
+def _get_foreign_landing_page(foreign_identifier):
+    url = f"https://{HOST}/photo/{foreign_identifier}"
+    final_url, page = None, None
+    response = delayed_requester.get(url)
+    if response:
+        final_url = response.url
+        page = html.document_fromstring(response.text)
+    return final_url, page
+
+
+def _get_image_info(media_data):
+    width = media_data.get('img_width')
+    height = media_data.get('img_height')
+    img_id = media_data.get('img_id')
+    image_url = f"{CDN}/{img_id}.jpg"
+    return image_url, width, height
+
+
+def _get_creator_data(page):
+    if len(page.find_class("author")) == 0:
+        return None, None
+    author_elem = page.find_class("author")[0]
+    creator = author_elem.text_content().strip()
+    creator_url = None
+    href = author_elem.attrib.get('href')
+    if href is not None:
+        creator_url = f"https://{HOST}{href}"
+    return creator, creator_url
+
+
+def _get_title(page):
+    """
+    Get the photo's title and transform it to title case, as shown on its page.
+    So for example, for "owl bird Photo" it returns "Owl Bird Photo".
+    """
+    title_str = page.xpath("//h1/span")[0].text_content()
```

Comment:
When the page design changes, this line may cause IndexError if there are no more spans inside the `<h1>` element. It is safer to use `try / except ` here.